### ISSUE TITLE
bug: .git/config remote URL corrupted with angle brackets by AI agent

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#1082]
## Summary

The remote origin URL in `.git/config` is written as `url = <https://github.com/rjmurillo/moq.analyzers>` instead of the correct `url = https://github.com/rjmurillo/moq.analyzers`. Git rejects this with:

```
fatal: protocol '<https' is not supported
```

This recurs at the start of each session.

## Root cause

An AI agent (Claude Code or a tool it invokes) generates a `git remote set-url` command with the URL in markdown autolink format. Example:

```
git remote set-url origin <https://github.com/rjmurillo/moq.analyzers>
```

The shell passes the angle brackets literally to git, which stores them in `.git/config`. The agent re-runs initialization logic each session, so the corruption recurs.

Investigation confirmed:
- No active git hooks in `.git/hooks/` write to `.git/config`.
- `.githooks/pre-commit` and `.githooks/pre-push` only invoke PowerShell linting; neither calls `git remote set-url`.
- No `includeIf` directive in `~/.gitconfig` conditionally rewrites the origin URL.
- `.serena/project.yml` line 18 contains an angle-bracket URL in a YAML comment (`<https://docs.python.org/...>`). This is cosmetic only and does not trigger a `git remote set-url` call. It is not the source of the corruption.
- The source is agent-generated shell commands that embed markdown autolink syntax (`<https://...>`) in git invocations.

## Local fix

Run once to repair:

```python
python3 -c "
c = open('.git/config').read()
open('.git/config', 'w').write(
    c.replace('url = <https://github.com/rjmurillo/moq.analyzers>',
              'url = https://github.com/rjmurillo/moq.analyzers')
)
"
```

Or with git directly after the repair:

```
git remote set-url origin https://github.com/rjmurillo/moq.analyzers
```

## Permanent fix

Audit all agent session initialization prompts and system instructions for any text that includes the repository URL in markdown autolink format (`<https://...>`). Replace all such occurrences with plain URLs. The specific candidates to check:

- Any `initial_instructions` or `initial_prompt` fields in `.serena/project.yml` or `.serena/` configs
- Any agent memory entries that reference the repo URL with angle brackets
- The global Claude instructions in `~/.claude/CLAUDE.md` or project-level `CLAUDE.md`

Optionally, add a `post-checkout` hook or session-init script that validates and repairs `.git/config` before any git operation:

```bash
#!/usr/bin/env bash
# .githooks/session-init
python3 -c "
c = open('.git/config').read()
if '<https://' in c:
    import re
    c = re.sub(r'url = <(https?://[^>]+)>', r'url = \1', c)
    open('.git/config', 'w').write(c)
    print('Repaired .git/config: removed angle brackets from remote URL')
"
```

## Acceptance criteria

- [ ] The specific agent prompt or tool that generates `git remote set-url origin <https://...>` is identified
- [ ] That prompt or tool is patched to use plain URLs
- [ ] The corruption does not recur after the patch
- [ ] A session-init repair script is added as a stopgap if root cause cannot be fully eliminated

## Related

Splits from #1080. See also the companion issue for the `.git/packed-refs` blank line corruption.

**Comment by rjmurillo** [comment]
## Recurrence source removed on `main` by #1215

The concrete driver of this corruption was the Codacy Cursor rule's MD034 "wrap bare URLs in `<angle brackets>`" auto-apply behavior (see #1084). That rule shipped in `.cursor/rules/codacy.mdc`, which PR #1215 (merged 2026-07-04) deleted. Verified: `origin/main` contains no `codacy.mdc`, and `.github/skills/moq-analyzers-failure-archaeology/SKILL.md:409` documents the guard *"Do not add AI-editor rules that auto-apply fixes after edits."*

With the auto-apply rule gone, the systemic source that rewrote `url = https://...` as `url = <https://...>` is removed.

The one-time local repair for anyone still holding a stale checkout remains valid:
```
git remote set-url origin https://github.com/rjmurillo/moq.analyzers
```
Closing as resolved (root cause removed on `main`).
